src/modules/sign/containers/GithubLoginContainer.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useContext, useEffect } from "react";
+import { useHistory } from "react-router-dom";
+import { StateContext } from "../../../App";
+import { axiosInstance } from "../components/axiosApi";
+import GitSignIn from "../components/gitSignIn";
+
+const logIn = async function (token, dispatch, history, isGitlab) {
```

Comment:
My thoughts about this login action.

Since the main difference between the Gitlab and Github flow is the endpoint `gitlab_auth` for Gitlab and `github_auth` for Github with everything else remaining the same. You could completely remove the `if-else` and duplicates by dynamically generating the auth endpoint and use the same code for both Github and Gitlab. I would do it as below:

```jsx
const logIn = async function (token, dispatch, history, isGitlab) {
    const authEndPoint = isGitlab ? 'gitlab_auth' : 'github_auth';
    const response = await axiosInstance.post(
        `http://mentorpaired-staging-backend.herokuapp.com/${authEndPoint}/`,
        { code: token }
     );
    // then reuse the same code since everything else is the same
}
```

Reply:
WOAH, this is a whole lot better!! 
I'll use this. Thanks.